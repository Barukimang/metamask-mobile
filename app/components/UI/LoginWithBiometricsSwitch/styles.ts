/* eslint-disable import/prefer-default-export */
import { fontStyles } from '../../../styles/common';
import { StyleSheet } from 'react-native';

export const createStyles = (colors: any) =>
  StyleSheet.create({
    screen: {
      flex: 1,
      backgroundColor: colors.background.default,
    },
    biometrics: {
      position: 'relative',
      marginTop: 20,
      marginBottom: 30,
    },
    biometryLabel: {
      flex: 1,
      fontSize: 16,
      color: colors.text.default,
      ...fontStyles.normal,
    },
    biometrySwitch: {
      position: 'absolute',
      top: 0,
      right: 0,
    },
  });
