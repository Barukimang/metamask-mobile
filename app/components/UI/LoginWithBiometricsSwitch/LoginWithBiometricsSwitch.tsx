import React, { useCallback } from 'react';
import { View, Switch, Text } from 'react-native';
import { mockTheme, useAppThemeFromContext } from '../../../util/theme';
import { strings } from '../../../../locales/i18n';
import AsyncStorage from '@react-native-community/async-storage';
import { TRUE, BIOMETRY_CHOICE_DISABLED } from '../../../constants/storage';
import { BIOMETRY_TYPE } from 'react-native-keychain';
import { createStyles } from './styles';
import { LOGIN_WITH_BIOMETRICS_SWITCH } from '../../../constants/test-ids';

interface Props {
  biometryType: BIOMETRY_TYPE;
  biometryEnabled: boolean;
}

const LoginWithBiometricsSwitch = ({
  biometryType,
  biometryEnabled,
}: Props) => {
  const { colors } = useAppThemeFromContext() || mockTheme;
  const styles = createStyles(colors);

  const updateBiometryChoice = useCallback(async () => {
    if (!biometryEnabled) {
      await AsyncStorage.setItem(BIOMETRY_CHOICE_DISABLED, TRUE);
    } else {
      await AsyncStorage.removeItem(BIOMETRY_CHOICE_DISABLED);
    }
  }, [biometryEnabled]);

  return (
    <View style={styles.biometrics} testID={LOGIN_WITH_BIOMETRICS_SWITCH}>
      <Text style={styles.biometryLabel}>
        {strings(`biometrics.enable_${biometryType.toLowerCase()}`)}
      </Text>
      <Switch
        onValueChange={updateBiometryChoice}
        value={biometryEnabled}
        style={styles.biometrySwitch}
        trackColor={{
          true: colors.primary.default,
          false: colors.border.muted,
        }}
        thumbColor={colors.white}
        ios_backgroundColor={colors.border.muted}
      />
    </View>
  );
};

export default React.memo(LoginWithBiometricsSwitch);
