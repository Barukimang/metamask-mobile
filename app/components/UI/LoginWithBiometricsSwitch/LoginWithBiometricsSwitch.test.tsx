import React from 'react';
import { shallow } from 'enzyme';
import LoginWithBiometricsSwitch from './LoginWithBiometricsSwitch';
import { BIOMETRY_TYPE } from 'react-native-keychain';
describe('LoginWithBiometricsSwitch', () => {
  // eslint-disable-next-line @typescript-eslint/no-empty-function
  const handleUpdate = (_biometricsEnabled: boolean) => {};
  it('should render correctly', () => {
    const wrapper = shallow(
      <LoginWithBiometricsSwitch
        biometryType={BIOMETRY_TYPE.FINGERPRINT}
        onUpdateBiometryChoice={handleUpdate}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
