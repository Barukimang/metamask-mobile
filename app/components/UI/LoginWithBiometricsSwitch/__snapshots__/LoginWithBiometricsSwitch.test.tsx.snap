// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginWithBiometricsSwitch should render correctly 1`] = `
<View
  style={
    Object {
      "marginBottom": 30,
      "marginTop": 20,
      "position": "relative",
    }
  }
  testID="login-with-biometrics-switch"
>
  <Text
    style={
      Object {
        "color": "#24272A",
        "flex": 1,
        "fontFamily": "EuclidCircularB-Regular",
        "fontSize": 16,
        "fontWeight": "400",
      }
    }
  >
    [missing "en.biometrics.enable_face" translation]
  </Text>
  <ForwardRef(Switch)
    ios_backgroundColor="#D6D9DC"
    onValueChange={[Function]}
    style={
      Object {
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
    trackColor={
      Object {
        "false": "#D6D9DC",
        "true": "#037DD6",
      }
    }
    value={false}
  />
</View>
`;
